homeassistant/components/homeassistant_sky_connect/config_flow.py

```diff
@@ -35,7 +35,12 @@ async def async_step_usb(self, discovery_info: usb.UsbServiceInfo) -> FlowResult
         manufacturer = discovery_info.manufacturer
         description = discovery_info.description
         unique_id = f"{vid}:{pid}_{serial_number}_{manufacturer}_{description}"
-        if await self.async_set_unique_id(unique_id):
+        if existing_entry := await self.async_set_unique_id(unique_id):
+            # Re-enable existing config entry which was disabled by USB unplug
+            if existing_entry.disabled_by == ConfigEntryDisabler.INTEGRATION:
+                await self.hass.config_entries.async_set_disabled_by(
+                    existing_entry.entry_id, None
+                )
```

Comment:
This will load the config entry. Is that OK, or should it be done by a task?

Reply:
We often await the reload when re-authenticating, so that should be fine.